### ISSUE TITLE
Jupyterlab is missing a way to specify options via the command line

### ISSUE THREAD

**Issue by adelcast** [jupyterlab/jupyterlab#5122]
In my use case, I am trying to modify JupyterLab to only enable an extension (jupterlab-hub) if the user is a an admin (I am using Jupyterhub). From jupyterhub, I can pass a True/False admin argument to the singleuserserver, which start jupyter....but I can't find a way to pass that info into the jupyterlab subsystem. I would have expected a --Jupyterlab.DisableExtension configuration option, or something similar, as far as I can tell, the only interface I have is via the page_config.json file. I could override --app-dir, but that sounds less than ideal since that approach will still rely on static files.

Not sure if there is a long term plan to add command line options to Jupterlab? Since its a nbextension, I see how this problem might be more generic...or maybe there is another approach that I am missing?

**Comment by blink1073** [comment]
Nice suggestion!  I agree we should think about how to allow `disabledExtensions`, `deferredExtensions`, and `uninstalled_core_extensions` be overridden by standard Jupyter config.

**Comment by bollwyvl** [comment]
Right, this would be needed for, say, an alternate file browser extension
to be distributed via pip/conda and disable the one in core.

This is complicated by custom app-dirs, which are a honking great idea, but
couldn't be targeted by hands-off config... But that's probably cool, as I
think that case will be targeted at the less-npm-interested.

Seems like an $app-dir/config.d would probably be needed for happy-path,
stock installation, and you kinda get what you get once you start using
something else.

Looking in $PREFIX, $HOME and $(cwd) is probably what people would expect,
though, but this could do some really wacky things...
